api/block/schema.py

```diff
@@ -28,22 +29,34 @@ class Arguments:
 
     @Auth.user_roles('Admin')
     def mutate(self, info, **kwargs):
-        validate_empty_fields(**kwargs)
-        get_office = Office.query.filter_by(id=kwargs['office_id']).first()
-        if not get_office:
-            raise GraphQLError("Office not found")
-        location = get_office.location.name
-        if location.lower() == 'nairobi':
-            block = BlockModel(**kwargs)
-            payload = {
-                'model': BlockModel, 'field': 'name', 'value':  kwargs['name']
-            }
-            with SaveContextManager(
-                block, 'Block', payload
-            ):
-                return CreateBlock(block=block)
-        else:
-            raise GraphQLError("You can only create block in Nairobi")
+        try:
+            validate_empty_fields(**kwargs)
+            get_office = Office.query.filter_by(id=kwargs['office_id']).first()
+            if not get_office:
+                raise GraphQLError("Office not found")
+            location = get_office.location.name
+            if location.lower() == 'nairobi':
+                block = BlockModel(**kwargs)
+                payload = {
+                    'model': BlockModel, 'field': 'name',
+                    'value':  kwargs['name']
+                }
+                query = Block.get_query(info)
+                result = query.filter(BlockModel.state == "active")
+                block_name = result.filter(
+                    func.lower(BlockModel.name) ==
+                    func.lower(kwargs['name'])).count()
+                if block_name > 0:
+                    ErrorHandler.check_conflict(self, kwargs['name'], 'Block')
+                with SaveContextManager(
+                    block, 'Block', payload
+                ):
+                    return CreateBlock(block=block)
+            else:
+                raise GraphQLError("You can only create block in Nairobi")
+        except exc.ProgrammingError:
+            raise GraphQLError("There seems to be a database connection error, \
```

Comment:
Is there a way you could avoid the back slash?

Reply:
By making the message shorter, that would be the only way @Bonifase

Reply:
Hey @C3real-kill3r , doe this ```
                raise GraphQLError(
                "There seems to be a database connection error" 
                ", contact your administrator for assistance")
               ``` 
solve the raised issue about the backslash?